### PR TITLE
[AST] Fix circular reference with implicit some

### PR DESCRIPTION
**Pull request by angela-laar**
getPotentiallyOpaqueGenericParams has a redundant call to check if declarations contain any opaque type representations. We don't need to check this here. Instead we can simply collect the result type representaions without the additional checks.

rdar://102279814

### CODE REVIEW

**Comment by angela-laar** [comment]
@swift-ci please smoke test

File: lib/AST/Decl.cpp [hunk 1]
```diff
@@ -3083,6 +3084,14 @@ TypeRepr *ValueDecl::getOpaqueResultTypeRepr() const {
     returnRepr = SD->getElementTypeRepr();
   }
 
+  return returnRepr;
+}
+
+TypeRepr *ValueDecl::getOpaqueResultTypeRepr() const {
+  TypeRepr *returnRepr = nullptr;
+
+  returnRepr = this->getResultTypeRepr();
```
Comment (hborla) [inline lib/AST/Decl.cpp:3093]: Nit: `TypeRepr *returnRepr` doesn't need to be initialized with `nullptr` first anymore, you can just write
`auto *returnRepr = getResultTypeRepr();`

**Review by hborla — APPROVED** [review]
Nice! One small comment inline

**Comment by angela-laar** [comment]
@swift-ci please smoke test

**Comment by angela-laar** [comment]
@swift-ci please smoke test

**Pull request merged**